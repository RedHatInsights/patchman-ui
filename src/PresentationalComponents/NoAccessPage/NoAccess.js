import { Main } from '@redhat-cloud-services/frontend-components/Main';
import React from 'react';
import Header from '../Header/Header';
import { intl } from '../../Utilities/IntlProvider';
import messages from '../../Messages';
import ErrorHandler from '../Snippets/ErrorHandler';
import propTypes from 'prop-types';

const NoAccess = ({ code }) => (
    <React.Fragment>
        <Header title={intl.formatMessage(messages.generalAppName)}/>
        <Main>
            <ErrorHandler code={code}  />
        </Main>
    </React.Fragment>
);

NoAccess.propTypes = {
    code: propTypes.number
};

export default NoAccess;
