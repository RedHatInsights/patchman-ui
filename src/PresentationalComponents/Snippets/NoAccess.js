import React from 'react';
import { withRouter } from 'react-router-dom';
import { LockIcon } from '@patternfly/react-icons';
import { Main } from '@redhat-cloud-services/frontend-components';
import { Bullseye, EmptyState, EmptyStateBody, EmptyStateIcon, EmptyStateVariant, Page, Title } from '@patternfly/react-core';
import { intl } from '../../Utilities/IntlProvider';
import messages from '../../Messages';

const NoAccessPage = () => {
    return (
        <Page>
            <Main>
                <Bullseye>
                    <EmptyState variant={EmptyStateVariant.large}>
                        <EmptyStateIcon icon={LockIcon} />
                        <Title headingLevel="h5" size="lg">
                            {intl.formatMessage(messages.statesRequiresPatchPermissions)}
                        </Title>
                        <EmptyStateBody>
                            {intl.formatMessage(messages.statesMinimumPatchPermissionsRequired)}
                        </EmptyStateBody>
                    </EmptyState>
                </Bullseye>
            </Main>
        </Page>
    );
};

export default withRouter(NoAccessPage);
