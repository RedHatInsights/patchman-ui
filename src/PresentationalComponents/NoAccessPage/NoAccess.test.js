import NoAccess from './NoAccess';
import toJson from 'enzyme-to-json';

describe('RegisterPage', () => {
    it('Should match the snapshot', () => {
        const wrapper = shallow(<NoAccess />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
