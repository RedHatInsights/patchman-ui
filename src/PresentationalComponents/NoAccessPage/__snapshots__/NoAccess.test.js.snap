// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterPage Should match the snapshot 1`] = `
<Fragment>
  <Header
    title="Patch"
  />
  <Connect(InternalMain)>
    <ErrorHandler />
  </Connect(InternalMain)>
</Fragment>
`;
